backend/database-router.js

```diff
@@ -1,50 +1,96 @@
 'use strict'
 
-const readConfig = () => {
-  const configString = process.env.GI_PERSIST_ROUTER_CONFIG
-  const config = JSON.parse(configString)
+import { resolve } from 'node:path'
+import { readFile } from 'node:fs/promises'
+import DatabaseBackend from './DatabaseBackend.js'
+import type { IDatabaseBackend } from './DatabaseBackend.js'
 
-  if (!config['*']) {
-    throw new Error('Fallback storage (*) is required')
+type Config = {
+  [string]: { name: string, options: Object }
+}
+type ConfigEntry = { name: string, options: Object }
```

Comment:
If the order of these type declarations is reversed, it can be simplified and DRYed:

```ts
type ConfigEntry = { name: string, options: Object }
type Config = { [string]: ConfigEntry }
```